### PR TITLE
WW-4923 Null check to avoid NPE when not passing a JDBC connection

### PR DESCRIPTION
**Pull request by zepernick**

### CODE REVIEW

**Comment by yasserzamani** [comment]
Nice catch :+1: You're right; that method always has checked if conn is null except there you discovered.

**Comment by lukaszlenart** [comment]
A test is failing:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.899 s <<< FAILURE! - in org.apache.struts2.views.jasperreports.JasperReportsResultTest
[ERROR] testConnClose(org.apache.struts2.views.jasperreports.JasperReportsResultTest)  Time elapsed: 2.81 s  <<< ERROR!
java.lang.NullPointerException
	at org.apache.struts2.views.jasperreports.JasperReportsResultTest.testConnClose(JasperReportsResultTest.java:68)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   JasperReportsResultTest.testConnClose:68 » NullPointer
```

File: plugins/jasperreports/src/main/java/org/apache/struts2/views/jasperreports/JasperReportsResult.java [hunk 1]
```diff
@@ -384,7 +384,10 @@ protected void doExecute(String finalLocation, ActionInvocation invocation) thro
             throw new ServletException(e.getMessage(), e);
         } finally {
             try {
-                conn.close();
+                if (conn != null) {
+                    // avoid NPE if connection was not used for the report
```
Comment (lukaszlenart) [inline plugins/jasperreports/src/main/java/org/apache/struts2/views/jasperreports/JasperReportsResult.java]: I would convert this comment into a log statement:
```java
LOG.debug("Closing connection used in report");
```

**Comment by yasserzamani** [comment]
> A test is failing

Yes they're my newly added tests to pass coverall :) I pushed one more commit which seems has fixed this test.

**Comment by lukaszlenart** [comment]
LGTM 👍

**Pull request merged**